Makefile

```diff
@@ -105,34 +96,11 @@ examples: consumer flask fastapi messaging
 
 .PHONY: package
 package:
-	python setup.py sdist
+	hatch build
 
 
 .PHONY: test
-test: deps
-	flake8
-	pydocstyle pact
-	coverage erase
-	tox
-	coverage report -m --fail-under=100
-
-.PHONY: venv
-venv:
-	@if [ -d "./.venv" ]; then echo "$(red).venv already exists, not continuing!$(sgr0)"; exit 1; fi
-	@type pyenv >/dev/null 2>&1 || (echo "$(red)pyenv not found$(sgr0)"; exit 1)
-
-	@echo "\n$(green)Try to find the most recent minor version of the major version specified$(sgr0)"
-	$(eval PYENV_VERSION=$(shell pyenv install -l | grep "\s\s$(PYTHON_MAJOR_VERSION)\.*" | tail -1 | xargs))
-	@echo "$(PYTHON_MAJOR_VERSION) -> $(PYENV_VERSION)"
-
-	@echo "\n$(green)Install the Python pyenv version if not already available$(sgr0)"
-	pyenv install $(PYENV_VERSION) -s
-
-	@echo "\n$(green)Make a .venv dir$(sgr0)"
-	~/.pyenv/versions/${PYENV_VERSION}/bin/python3 -m venv ${CURDIR}/.venv
-
-	@echo "\n$(green)Make it 'available' to pyenv$(sgr0)"
-	ln -sf ${CURDIR}/.venv ~/.pyenv/versions/${PROJECT}
-
-	@echo "\n$(green)Use it! (populate .python-version)$(sgr0)"
-	pyenv local ${PROJECT}
+test:
+	hatch run all
```

Comment:
fails on lint, as noted in the ci scripts

Reply:
```console
🕙15:18:21 ❯ hatch run test:all
───────────────────────────────────────────────────────────────────── test.py3.8 ─────────────────────────────────────────────────────────────────────
/bin/sh: tests: command not found

```

PS not sure why its saying `py3.8`, my currently installed versions are as below. Is that failing because I don't have python 3.8.

```
❯ pyenv versions 
  system
  2.7.18
* 3.11.4 (set by /Users/yousaf.nabi/.pyenv/version)

```

`hatch run test` works grand and all the unit tests passing. can see my wheel built the dist folder.

How would one directly install the egg in an external project to test?

Reply:
Within hatch, I have defined two sets of venvs. The default venv is the one used interactively by hatch during regular development work, and will use whatever (compatible) Python you have as default.

I have also defined a `test` environment which is itself a matrix of Python version:

https://github.com/pact-foundation/pact-python/blob/324a624149e909ee8174537407db3d9d190d640e/pyproject.toml#L115-L121

When running `hatch run test:all`, it should setup a venv for each Python version in the matrix and run the test suite. Hatch, to my knowledge, automatically discovers Python versions in your system.

If you don't have Python 3.8 at all, I don't know how this ran for you. Do you have it installed separately to `pyenv`, perhaps with `brew install python@3.8` or maybe `conda` if it supports this? I don't believe hatch will automatically download and install versions of Python automatically, but I might be wrong.

Reply:
ahh yeah there will more than likely be a python@3.8 required from various homebrew packages, not managed by pyenv